src/overlays/actors/ovl_Boss_Ganon2/z_boss_ganon2.c

```diff
@@ -278,116 +304,3125 @@ static ColliderJntSphInit D_80907014 = {
+        }
+    }
+
+    effect = effects;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_80903F38.s")
+    for (i = 0; i < 100; i++, effect++) {
+        if (effect->type == 2) {
+            if (spCD == 0) {
+                BossGanon2_SetObjectSegment(NULL, globalCtx, OBJECT_GEFF, true);
+                spCD++;
+            }
+            Matrix_Translate(effect->position.x, effect->position.y, effect->position.z, MTXMODE_NEW);
+            Matrix_Scale(effect->scale, effect->scale, effect->scale, MTXMODE_APPLY);
+            Matrix_RotateY(effect->unk_38.z, MTXMODE_APPLY);
+            Matrix_RotateX(effect->unk_38.y, MTXMODE_APPLY);
+            Matrix_RotateZ(effect->unk_38.x, MTXMODE_APPLY);
+            gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_boss_ganon2.c", 6179),
+                      G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+            gSPDisplayList(POLY_OPA_DISP++, D_06000EA0);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_80904108.s")
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_boss_ganon2.c", 6185);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_80904340.s")
+void func_80906538(BossGanon2* this, u8* shadowTexture, f32 arg2) {
+    s16 temp_t0;
+    s16 temp_v0;
+    s16 temp_a3;
+    s16 phi_v1;
+    s16 phi_a1;
+    s16 i;
+    f32 lerpx;
+    s16 j;
+    f32 lerpy;
+    f32 lerpz;
+    Vec3f sp70;
+    Vec3f sp64;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_8090464C.s")
+    for (i = 0; i < 15; i++) {
+        //! @bug j is not initialized if arg2 == 0.0f, causing undefined behavior.
+        if ((arg2 == 0.0f) || ((j = D_809071CC[i]) >= 0)) {
+            if (arg2 > 0.0f) {
+                lerpx = this->unk_234[i].x + (this->unk_234[j].x - this->unk_234[i].x) * arg2;
+                lerpy = this->unk_234[i].y + (this->unk_234[j].y - this->unk_234[i].y) * arg2;
+                lerpz = this->unk_234[i].z + (this->unk_234[j].z - this->unk_234[i].z) * arg2;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_80904818.s")
+                sp70.x = lerpx - this->actor.world.pos.x;
+                sp70.y = lerpy - this->actor.world.pos.y + 76.0f + 30.0f + 30.0f + 100.0f;
+                sp70.z = lerpz - this->actor.world.pos.z;
+            } else {
+                sp70.x = this->unk_234[i].x - this->actor.world.pos.x;
+                sp70.y = this->unk_234[i].y - this->actor.world.pos.y + 76.0f + 30.0f + 30.0f + 100.0f;
+                sp70.z = this->unk_234[i].z - this->actor.world.pos.z;
+            }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_809049A0.s")
+            Matrix_MultVec3f(&sp70, &sp64);
+            sp64.x *= 0.2f;
+            sp64.y *= 0.2f;
+            temp_a3 = sp64.x + 32.0f;
+            temp_t0 = (s16)sp64.y << 6;
```

Comment:
Can this be a multiplication?

Reply:
I'm not sure, do you have any suggestions for me to try? Sda has something similar. https://github.com/zeldaret/oot/blob/master/src/overlays/actors/ovl_En_Sda/z_en_sda.c#L197

Reply:
Should be *64

Reply:
Yep, that works. I guess earlier I was trying to multiply it as a float haha.